### ISSUE TITLE
Changing particle gizmo color doesn't have effect until restart

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#56403]
### Godot version

bc2445d

### System information

W10

### Issue description

When you change color for Particle 3D Gizmo, you need to restart editor for the change to have effect. Might be true for other gizmos too.

### Steps to reproduce

1. Create GPUParticles3D node
2. Change Editor Settings -> Editors -> 3D Gizmos -> Particles to different color
3. Notice it doesn't have effect
4. Restart editor
5. Notice it has effect

**Comment by viksl** [comment]
Yep, it's not just particles this is in general, new 3d gizmo color doesn't appear to change until you restart the editor.

**Comment by Calinou** [comment]
> EDIT: Also the color changes don't seem to propagate to debug view when you hit the play button, for example collision shapes change color in 3.5 even in the game window but they don't change in 4.0, it only changes in the editor view.

Debug collision shape colors are set in the *project* settings, not editor settings, so this is an unrelated issue. Please open a separate issue with a minimal reproduction project attached.

**Comment by viksl** [comment]
@Calinou Oh, no need to, I'm an idiot, thanks for reminding me I set it up properly in 3.5 and set it wrong in 4.0. The part about debug not working is false, it works.

The issues with the gizmos not changing colors until restart is confirmed and true as far as I can tell.
